### PR TITLE
Core death improvements

### PR DESCRIPTION
**Pull request by Konijima**
- Fixed a glitch with regdoll/dead animation.
- Added screenEffect helper class and enum.
- Changed the death screen text, size and position.
- Changed the respawn to use a keypress.
- Re-worked the overall death effects.

### CODE REVIEW

**Comment by Konijima** [comment]
Without the `player.spawn` line removed here [727ec86](https://github.com/Stuyk/altv-athena/pull/334/commits/727ec86f9bb0c1af59574fb99a1611fed58ac335) there seem to be a problem with respawning from a real death.

File: src/core/client/utility/screenEffect.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import * as natives from "natives";
+import { SCREEN_EFFECTS } from "../../shared/enums/ScreenEffects";
+
+export abstract class ScreenEffect {
```
Comment (Stuyk) [inline src/core/client/utility/screenEffect.ts:4]: These are all static; so let's try to move away from this and go towards consts.

See: #299

File: src/core/shared/configurations/shared.ts [hunk 2]
```diff
@@ -1,6 +1,8 @@
 export const SHARED_CONFIG = {
     // Default Respawn Time for Player
     RESPAWN_TIME: 25000,
+    // Default Respawn Key
+    RESPAWN_KEY: 88, // X
```
Comment (Stuyk) [inline src/core/shared/configurations/shared.ts:5]: This key should be configured in Death Plugin since it will be meaningless if the plugin is removed.

**Review by Stuyk — CHANGES_REQUESTED** [review]

File: src/core/client/utility/screenEffect.ts (hunk 1 above)
Comment (Konijima) [reply src/core/client/utility/screenEffect.ts]: [f16bd15](https://github.com/Stuyk/altv-athena/pull/334/commits/f16bd15982cc5d7d7194af1030db89a71a212ef5)

File: src/core/shared/configurations/shared.ts (hunk 2 above)
Comment (Konijima) [reply src/core/shared/configurations/shared.ts]: [b6af687](https://github.com/Stuyk/altv-athena/pull/334/commits/b6af687a6576b49de3b419ea034a45599de7cdec)

**Review by Stuyk — APPROVED** [review]

**Pull request merged**